libcxx/include/__charconv/from_chars_floating_point.h

```diff
@@ -0,0 +1,73 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___CHARCONV_FROM_CHARS_FLOATING_POINT_H
+#define _LIBCPP___CHARCONV_FROM_CHARS_FLOATING_POINT_H
+
+#include <__assert>
+#include <__charconv/chars_format.h>
+#include <__charconv/from_chars_result.h>
+#include <__config>
+#include <__system_error/errc.h>
+#include <cstddef>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+_LIBCPP_PUSH_MACROS
+#include <__undef_macros>
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+#if _LIBCPP_STD_VER >= 17
+
+template <class _Fp>
+struct __from_chars_result {
+  _Fp __value;
+  ptrdiff_t __n;
+  errc __ec;
+};
+
+template <class _Fp>
+_LIBCPP_EXPORTED_FROM_ABI [[gnu::pure]] __from_chars_result<_Fp> __from_chars_floating_point(
+    [[clang::noescape]] const char* __first, [[clang::noescape]] const char* __last, chars_format __fmt);
```

Comment:
```suggestion
template <class _Fp>
_LIBCPP_EXPORTED_FROM_ABI [[__gnu__::__pure__]] __from_chars_result<_Fp> __from_chars_floating_point(
    [[_Clang::__noescape__]] const char* __first, [[clang::noescape]] const char* __last, chars_format __fmt);
```
Also, `[[clang::noescape]]` should be behind a macro, since GCC doesn't support it.

@ldionne What's blocking re-enabling clang tidy? This would have been caught.

Reply:
I discussed using macros with Louis. He'll work on a patch to document how we want to use exported dylib functions; that patch should introduce that macro.

Reply:
For now I removed the pure, some compilers seem not to like it. I'll investigate this later.

Reply:
> @ldionne What's blocking re-enabling clang tidy? This would have been caught.

We need to update the Docker image for that, and that's blocked on https://github.com/llvm/llvm-project/pull/110303 which is blocked on https://github.com/llvm/llvm-project/pull/110554, which is awaiting review.